### PR TITLE
Added Google Search Engine Clone

### LINKED ISSUE #403: Add Google Search Clone
**Project title**
Google Search Engine Clone

**Project Description**
Google Search Engine Clone using React Js with the help of Redux and Google Context API , it works like a real Google search engine if you search something it will give you all results 

**Stack**:  
And in order to tick the check box just put x inside them for example - [x] like this. Please delete options that are not relevant.

- [x] Html
- [x] CSS
- [x] JavaScript
- [x] ReactJS
- [x] Node JS
- [x] Redux

### PR DESCRIPTION
**Pull request by gurjeetsinghvirdee**
<hr>

## Description 📜

Google Search Engine Clone using React js with the help of Google CONTEXT API

Fixes #403 

<hr>

## Type of change 📝

Please delete options that are not relevant.And in order to tick the check box just but x inside them for example [x] like this

- [x] Add a new project

<hr>

## Domain of Contribution 📊

Please delete options that are not relevant.And in order to tick the check box just but x inside them for example [x] like this

- [x] Project using ReactJs

<hr>
 
## Checklist ✅

Please delete options that are not relevant.And in order to tick the check box just but x inside them for example [x] like this

- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GSSoC'22 contributor

<hr>
 
## Screenshots / Gif (Optional) 📸
Screenshot (793)

Screenshot (794)

Screenshot (795)

Screenshot (796)

### CODE REVIEW

**Review by meinhoonharsh — COMMENTED** [review]
All things are good to get merged
@khushi-purwar It is a React Application, If we are merging Applications based on stack other than core HTML, CSS and JS then this PR can be merged

File: Google-Search-Engine-Clone/.eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...ojectKart\\Google-Search-Engine-Clone\\src\\SearchPage.js",["34","35"],"E:\\GSSo'C 22\\WebDev-ProjectKart\\Google-Search-Engine-Clone\\src\\Home.js",["36"],"E:\\GSSo'C 22\\WebDev-ProjectKart\\Google-Search-Engine-Clone\\src\\response.js",["37"],"E:\\GSSo'C 22\\WebDev-ProjectKart\\Google-Search-Engine-Clone\\src\\useGoogleSearch.js",[],"E:\\GSSo'C 22\\WebDev-ProjectKart\\Google-Search-Engine-Clone\\src\\config.js",[],{"ruleId":"38","severity":1,"message":"39","line":5,"column":8,"nodeType":"40","messageId":"41","endLine":5,"endColumn":16},{"ruleId":"38","severity":1,"message":"42","line":16,"column":22,"nodeType":"40","messageId":"41","endLine":16,"endColumn":30},{"ruleId":"38","severity":1,"message":"43","line":3,"column":10,"nodeType":"40","messageId":"41","endLine":3,"endColumn":14},{"ruleId":"44","severity":1,"message":"45","line":1,"column":1,"nodeType":"46","endLine":337,"endColumn":4},"no-unused-vars","'Response' is defined but never used.","Identifier","unusedVar","'dispatch' is assigned a value but never used.","'Link' is defined but never used.","import/no-anonymous-default-export","Assign object to a variable before exporting as module default","ExportDefaultDeclaration"]
```
Comment (meinhoonharsh) [inline Google-Search-Engine-Clone/.eslintcache:1]: `.eslintcache` just add this inside `.gitignore`

**Review by meinhoonharsh — CHANGES_REQUESTED** [review]

**Comment by khushi-purwar** [comment]
> All things are good to get merged @khushi-purwar It is a React Application, If we are merging Applications based on stack other than core HTML, CSS and JS then this PR can be merged

Yes, we are accepting react application

**Review by khushi-purwar — APPROVED** [review]

**Comment by Kushal997-das** [comment]
@khushi-purwar Please remove the tag from this PR as this PR exits in multiple repos

[repos1](https://github.com/surajm-333/Ace-The-FrontEnd/pull/103)
[repos2](https://github.com/Ayush7614/Bundli-Frontend/pull/558)
[repos3](https://github.com/Rayman-Sodhi/Clone-IT/pull/880)

**Pull request merged**